tests/rptest/tests/upgrade_test.py

```diff
@@ -311,3 +313,146 @@ def has_expected_topics():
             return num_topics == num_expected
 
         wait_until(has_expected_topics, timeout_sec=30, backoff_sec=1)
+
+
+class AdminRpcUpgradeTest(EndToEndTest):
+    def setUp(self):
+        self.initial_version = MixedVersionWorkloadRunner.PRE_SERDE_VERSION
+        install_opts = InstallOptions(version=self.initial_version)
+        # This test is sensitive to leadership changes, so make sure we only
+        # change leaders intentionally.
+        extra_rp_conf = dict(enable_leader_balancer=False)
+        self.start_redpanda(num_nodes=4,
+                            install_opts=install_opts,
+                            extra_rp_conf=extra_rp_conf)
+
+    def run_admin_rpcs(self, src_node, dst_node):
+        """
+        Performs several admin operations. This exercises requests sent by
+        the controller, as the RPCs are expected to land at the controller
+        leader 'dst_node' and they must be forwarded from 'src_node'.
+        """
+        # Create new topic partitions to help ensure node decommissioning takes
+        # time and we can slip in a recommission.
+        spec = TopicSpec(name=rand_with_prefix("topic"),
+                         partition_count=2,
+                         replication_factor=3)
+        self.client().create_topic(spec)
+
+        admin = self.redpanda._admin
+        dst_id = self.redpanda.idx(dst_node)
+        wait_until(lambda: admin.transfer_leadership_to(namespace="redpanda",
+                                                        topic="controller",
+                                                        partition=0,
+                                                        target_id=dst_id),
+                   timeout_sec=60,
+                   backoff_sec=1)
+        admin.await_stable_leader(namespace="redpanda",
+                                  topic="controller",
+                                  partition=0)
+        # Avoid nodes of interest by operating on the back node.
+        back_node = self.redpanda.nodes[-1]
+        back_id = self.redpanda.idx(back_node)
```

Comment:
I may be missing something here but does the MixedVersionWorkloadRunner never select the back node? What I'm confused by is why strictly selecting the last node is preferred over any others?

Reply:
Yeah, the mixed version runner runs workloads between the first two nodes deterministically, exercising a specific direction of RPC traffic (e.g. from old version to new version, or vice versa). Added a comment.